x-pack/metricbeat/module/meraki/wireless_device_channel_utilization/wireless_device_channel_utilization.go

```diff
@@ -0,0 +1,149 @@
+		return nil, err
+	}
+
+	logger.Debugf("loaded config: %v", config)
+	client, err := meraki_api.NewClientWithOptions(config.BaseURL, config.ApiKey, config.DebugMode, "Metricbeat Elastic")
+	if err != nil {
+		logger.Error("creating Meraki dashboard API client failed: %w", err)
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+		logger:        logger,
+		client:        client,
+		organizations: config.Organizations,
+	}, nil
+}
+
+// Fetch method implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
+	for _, org := range m.organizations {
+
+		devices, err := meraki.GetDevices(m.client, org)
+		if err != nil {
+			return err
+		}
+
+		res, err := m.client.Devices.GetOrganizationWirelessDevicesChannelUtilizationByDevice(org, &meraki_api.GetOrganizationWirelessDevicesChannelUtilizationByDeviceQueryParams{})
+		if err != nil {
+			return fmt.Errorf("GetOrganizationWirelessDevicesChannelUtilizationByDevice failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+		}
+
+		//debug
+		//fmt.Printf("\n #DEBUG Devices.GetOrganizationWirelessDevicesChannelUtilizationByDevice; [%d] %s.", res.StatusCode(), res.Body())
+
+		var wirelessDevices WirelessDevicesChannelUtilizationByDevice
+		err = json.Unmarshal(res.Body(), &wirelessDevices)
+		if err != nil {
+			return fmt.Errorf("device_network_health_channel_utilization json umarshal failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+		}
+
+		reportWirelessDeviceChannelUtilization(reporter, org, devices, wirelessDevices)
+
+	}
+
+	return nil
+}
+
+func reportWirelessDeviceChannelUtilization(reporter mb.ReporterV2, organizationID string, devices map[meraki.Serial]*meraki.Device, wirelessDevices WirelessDevicesChannelUtilizationByDevice) {
+
+	metrics := []mapstr.M{}
+
+	for _, wirelessDevice := range wirelessDevices {
+
+		if device, ok := devices[meraki.Serial(wirelessDevice.Serial)]; ok {
+
+			metric := mapstr.M{
+				"wireless.device.address":      device.Address,
+				"wireless.device.firmware":     device.Firmware,
+				"wireless.device.imei":         device.Imei,
+				"wireless.device.lan_ip":       device.LanIP,
+				"wireless.device.location":     device.Location,
+				"wireless.device.mac":          device.Mac,
+				"wireless.device.model":        device.Model,
+				"wireless.device.name":         device.Name,
+				"wireless.device.network_id":   device.NetworkID,
+				"wireless.device.notes":        device.Notes,
+				"wireless.device.product_type": device.ProductType,
+				"wireless.device.serial":       device.Serial,
+				"wireless.device.tags":         device.Tags,
+			}
+
+			for _, v := range wirelessDevice.ByBand {
+				metric[fmt.Sprintf("wireless.device.channel.utilization.band_%s.wifi.percentage", v.Band)] = v.Wifi.Percentage
```

Comment:
we might want to do a `strings.ReplaceAll(v.Band, ".", "_")` on these so that e.g. "2.4" doesn't get split into two levels.

Reply:
actually there's a convenience function in libbeat for this - `Dedot` in libbeat.common

Reply:
Thank you, I will use that method.

Reply:
Fixed with dedot